client/state/account-recovery/settings/reducer.js

```diff
@@ -1,33 +1,137 @@
 /**
+ * External dependencies
+ */
+import { combineReducers } from 'redux';
+
+/**
  * Internal dependencies
  */
 import { createReducer } from 'state/utils';
 
 import {
-	ACCOUNT_RECOVERY_SETTINGS_FETCH,
 	ACCOUNT_RECOVERY_SETTINGS_FETCH_SUCCESS,
-	ACCOUNT_RECOVERY_SETTINGS_FETCH_FAILED,
+
+	ACCOUNT_RECOVERY_SETTINGS_UPDATE,
+	ACCOUNT_RECOVERY_SETTINGS_UPDATE_SUCCESS,
+	ACCOUNT_RECOVERY_SETTINGS_UPDATE_FAILED,
+
+	ACCOUNT_RECOVERY_SETTINGS_DELETE,
+	ACCOUNT_RECOVERY_SETTINGS_DELETE_SUCCESS,
+	ACCOUNT_RECOVERY_SETTINGS_DELETE_FAILED,
 } from 'state/action-types';
 
-const reducer = createReducer( {}, {
-	[ ACCOUNT_RECOVERY_SETTINGS_FETCH ]: ( state ) => ( {
+const isUpdating = createReducer( {}, {
+	[ ACCOUNT_RECOVERY_SETTINGS_UPDATE ]: ( state, { target } ) => ( {
 		...state,
-		isFetching: true,
+		[ target ]: true,
 	} ),
-
-	[ ACCOUNT_RECOVERY_SETTINGS_FETCH_SUCCESS ]: ( state, { email, email_validated, phone, phone_validated } ) => ( {
+	[ ACCOUNT_RECOVERY_SETTINGS_UPDATE_SUCCESS ]: ( state, { target } ) => ( {
 		...state,
-		email,
-		emailValidated: email_validated,
-		phone,
-		phoneValidated: phone_validated,
-		isFetching: false,
+		[ target ]: false,
 	} ),
+	[ ACCOUNT_RECOVERY_SETTINGS_UPDATE_FAILED ]: ( state, { target } ) => ( {
+		...state,
+		[ target ]: false,
+	} ),
+} );
 
-	[ ACCOUNT_RECOVERY_SETTINGS_FETCH_FAILED ]: ( state ) => ( {
+const isDeleting = createReducer( {}, {
+	[ ACCOUNT_RECOVERY_SETTINGS_DELETE ]: ( state, { target } ) => ( {
 		...state,
-		isFetching: false,
+		[ target ]: true,
+	} ),
+	[ ACCOUNT_RECOVERY_SETTINGS_DELETE_SUCCESS ]: ( state, { target } ) => ( {
+		...state,
+		[ target ]: false,
+	} ),
+	[ ACCOUNT_RECOVERY_SETTINGS_DELETE_FAILED ]: ( state, { target } ) => ( {
+		...state,
+		[ target ]: false,
 	} ),
 } );
 
-export default reducer;
+const convertPhoneResponse = ( phoneResponse ) => {
+	if ( ! phoneResponse ) {
+		return null;
+	}
+
+	const {
+		country_code,
+		country_numeric_code,
+		number,
+		number_full,
+	} = phoneResponse;
+
+	return {
+		countryCode: country_code,
+		countryNumericCode: country_numeric_code,
+		number: number,
+		numberFull: number_full,
+	};
+};
+
+const convertEmailResponse = ( emailResponse ) => {
```

Comment:
Nice and concise :)
There is no plan to add more logic, but as refactoring the `data` reducer in 75b2fe39319f01ee8256e780b8c2d1ad5868f46f, it now uses in two places so I think it still worths a function.